src/platform/w3dengine/client/w3ddisplay.cpp

```diff
@@ -933,9 +1673,594 @@ bool W3DDisplay::Is_LetterBox_Fading()
+
+        Coord3D pos;
+        g_theTacticalView->Get_Position(&pos);
+        str1.Format(L"Camera zoom: %g, pitch: %g/%g, yaw: %g, pos: %g, %g, %g, FOV: %g\n"
+                    L"       Height above ground: %g Terrain height: %g",
+            g_theTacticalView->Get_Zoom(),
+            g_theTacticalView->Get_Pitch(),
+            g_theTacticalView->Get_FX_Pitch(),
+            g_theTacticalView->Get_Angle(),
+            pos.x,
+            pos.y,
+            pos.z,
+            g_theTacticalView->Get_Field_Of_View(),
+            g_theTacticalView->Get_Current_Height_Above_Ground(),
+            g_theTacticalView->Get_Terrain_Height_Under_Camera());
+        m_displayStrings[5]->Set_Text(str1);
+
+        str1.Format(L"States: ");
+
+        if (g_theKeyboard->Is_Shift()) {
+            str1.Concat(L"Shift(");
+
+            if ((g_theKeyboard->Get_Modifiers() & Keyboard::MODIFIER_LSHIFT) != 0) {
+                str1.Concat(L"L");
+            }
+
+            if ((g_theKeyboard->Get_Modifiers() & Keyboard::MODIFIER_RSHIFT) != 0) {
+                str1.Concat(L"R");
+            }
+
+            str1.Concat(L") ");
+        }
+
+        if (g_theKeyboard->Is_Ctrl()) {
+            str1.Concat(L"Ctrl(");
+            if ((g_theKeyboard->Get_Modifiers() & Keyboard::MODIFIER_LCTRL) != 0) {
+                str1.Concat(L"L");
+            }
+            if ((g_theKeyboard->Get_Modifiers() & Keyboard::MODIFIER_RCTRL) != 0) {
+                str1.Concat(L"R");
+            }
+            str1.Concat(L") ");
+        }
+
+        if (g_theKeyboard->Is_Alt()) {
+            str1.Concat(L"Alt(");
+
+            if ((g_theKeyboard->Get_Modifiers() & Keyboard::MODIFIER_LALT) != 0) {
+                str1.Concat(L"L");
+            }
+
+            if ((g_theKeyboard->Get_Modifiers() & Keyboard::MODIFIER_RALT) != 0) {
+                str1.Concat(L"R");
+            }
+            str1.Concat(L") ");
+        }
+
+        MouseIO *status = g_theMouse->Get_Mouse_Status();
+
+        if (status->left_state) {
+            str1.Concat(L"LMB ");
+        }
+
+        if (status->middle_state) {
+            str1.Concat(L"MMB ");
+        }
+
+        if (status->right_state) {
+            str1.Concat(L"RMB ");
+        }
+
+        const Object *object = nullptr;
+        Drawable *d;
+#ifdef GAME_DEBUG_STRUCTS
+        d = g_theTacticalView->Pick_Drawable((ICoord2D *)0x00E1B118, false, PICKTYPE_UNK2);
+
+#else
+        DrawableID id = g_theInGameUI->Get_Moused_Over_Drawable_ID();
+        d = g_theGameClient->Find_Drawable_By_ID(id);
+#endif
+        if (id) {
+            object = d->Get_Object();
+        }
+
+        if (object) {
+            str2.Format(L"Moused over object: %S (%d) ", object->Get_Template()->Get_Name().Str(), object->Get_ID());
+            str1.Concat(str2);
+        } else {
+            str1.Concat(L"Moused over object: TERRAIN ");
+        }
+
+        m_displayStrings[6]->Set_Text(str1);
```

Comment:
We need an enum for these indexes.
```
DISPLAY_STRING_OBJECT = 6,
DISPLAY_STRING_MOUSE_POSITION = 7,
```

and so on.

Reply:
Perhaps within function if so, i believe the array is mean as general purpose